### PR TITLE
DSDEGP-1626: Only check expected output cycles

### PR DESCRIPTION
**Pull request by jacarey**
DSDEGP-1626

This PR fixes the bug described here: https://gatkforums.broadinstitute.org/gatk/discussion/comment/42610

This ensures that cbcls are only checked for output cycles.
----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/illumina/CheckIlluminaDirectory.java [hunk 1]
```diff
@@ -168,9 +168,8 @@ protected int doWork() {
                     filterFileMap.put(fileToTile(filterFile.getName()), filterFile);
                 }
 
-                final OutputMapping outputMapping = new OutputMapping(readStructure);
-
-                final CbclReader reader = new CbclReader(cbcls, filterFileMap, readStructure.readLengths, tiles.get(0), locs, outputMapping.getOutputCycles(), true);
+                final CbclReader reader = new CbclReader(cbcls, filterFileMap, outputMapping.getOutputReadLengths(),
```
Comment (pshapiro4broad) [inline src/main/java/picard/illumina/CheckIlluminaDirectory.java]: It looks like the significant change here is replacing `readStructure.readLengths` with `outputMapping.getOutputReadLengths()`. Why did you remove the variable `expectedCycles` which was caching `outputMapping.getOutputCycles()` as well?

**Comment by jacarey** [comment]
Adding test data that would make this tests possible is .... complicated.

File: src/main/java/picard/illumina/CheckIlluminaDirectory.java (hunk 1 above)
Comment (jacarey) [reply src/main/java/picard/illumina/CheckIlluminaDirectory.java]: I'm caching the full outputmapping now. It is one extra get call now. Not sure its worth caching that as well.

Comment (pshapiro4broad) [reply src/main/java/picard/illumina/CheckIlluminaDirectory.java]: Well, it's not a simple getter (it calls `ReadStructure.getCycles()`), but I'll leave it up to you if it's worth caching or not.

**Comment by alecw** [comment]
Hi @sooheelee , I would think you could use any run folder, and simply replace the "natural" read structure with something unnatural but compatible, e.g. make the last template read 1 base shorter, and add a 1 base skip at the end.

**Comment by sooheelee** [comment]
Thanks for the clarification @alecw. I thought perhaps it was easiest to get some data from the user but what you say clarifies that it would be just as easy to modify existing data.

**Comment by jacarey** [comment]
I have tested it with real data. The issue is setting this up to be run properly with testdata for a unit test.
